server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -291,7 +302,7 @@ private static HitContext prepareNonNestedHitContext(
             Source source = Source.lazy(lazyStoredSourceLoader(profiler, subReaderContext, subDocId));
             return new HitContext(hit, subReaderContext, subDocId, Map.of(), source, rankDoc);
         } else {
-            SearchHit hit = new SearchHit(docId, id);
+            SearchHit hit = new SearchHit(docId, id, null);
```

Comment:
NIT: Why inline the overload here? Was this intentional?

Reply:
It's from back in the complicated-solution times. Dropped it. Thanks !